DEBUGGING.md

```diff
@@ -0,0 +1,71 @@
+# Debugging the OpenTofu code
+
+There are various ways to debug the OpenTofu code. There is no ultimate "right" answer, this document intends to collect some of those ways. The order of debugging techniques is completely random.
+
+If you would like to contribute to this debugging guide, please [create a GitHub issue](https://github.com/opentofu/opentofu/issues/new/choose) and propose the enhancement. After that, you can create a pull request and reference this issue in your PR.
+
+For further information on contributing to the code, please refer to the [CONTRIBUTING.md](./CONTRIBUTING.md) file.
+
+<!-- TOC -->
+
+- [Using the debug-opentofu script](#using-the-debug-opentofu-script)
+- [Using spew](#using-spew)
+- [Using VsCode](#using-vscode)
+
+<!-- /TOC -->
+
+## Using the debug-opentofu script
+
+[debug-opentofu](./scripts/debug-opentofu) is a helper script to launch OpenTofu inside the ["dlv" debugger](https://github.com/go-delve/delve), configured to await a remote debugging connection on port 2345. For more details on how to use this script, please refer to the documentation at the beginning of this script.
+
+## Using spew
+
+[Go-spew](https://github.com/davecgh/go-spew) implements a deep pretty printer for Go data structures to aid in debugging. If you prefer to use println debugging, spew.Dump might be helpful.
```

Comment:
```suggestion
[Go-spew](https://github.com/davecgh/go-spew) implements a deep pretty printer for Go data structures to aid in debugging. If you prefer to use println debugging, `spew.Dump` might be helpful.
```

Reply:
Direct commits from GitHub UI doesn't comply with DCO, so I `commit --amend --sign-off` this commit, FYI.